dask/array/tests/test_cupy.py

```diff
@@ -815,6 +815,20 @@ def test_sfqr(m, n, chunks, error_type):
             q, r = da.linalg.sfqr(data)
 
 
+def test_sparse_hstack_vstack_csr():
+    pytest.importorskip("cupyx")
+    x = cupy.array([2.0, 1.0, 5.0, 2.0, 1.0], dtype=cupy.float32)
+    x = cupy.expand_dims(x, axis=0)
```

Comment:
Maybe it is worth having a proper 2-D matrix so that both dimensions are tested?

```suggestion
    x = cupy.arange(24, dtype=cupy.float32).reshape(4, 6)
```

Reply:
Any thoughts on this @cjnolet?

Reply:
@jakirkham, unfortunately, this PR is a little lower in priority than some of the other PRs I've got in the mix so I have not had a great amount of time to look at this today.

Reply:
That being said, if you think you are able to take this across the finish line, I wouldn't have any complaints accepting commits to this branch.

Reply:
Sure I just didn't know if there was a reason to not accept this suggestion. It sounds like there isn't one.

Reply:
Ah, I see. I generally prefer to make the code change locally only because I like to run the tests on my workstation before checking them in instead of waiting for CI to run them. I've only really accepted suggestions for documentation. I don't mind accepting it if you prefer, but my personal preference would still be for a local verification that both hstack and vstack are actually being called (unless you saying you've already done this, then please excuse my ranting! ). 

There is also a pending issue with the slicing that Dask.Array uses because cupy's csr/csc matrices don't support all the slicing options that scipy's corresponding arrays support. An issue still needs to be opened about the latter case.

Reply:
Interesting. Was not aware there was a workflow involved. Yeah whatever you want to do here is fine with me.

Just wanted to make sure we were using a test case that was exercising concatenation along both dimensions.

I mean this line works...

```python
In [1]: import cupy                                                             

In [2]: cupy.arange(24, dtype=cupy.float32).reshape(4, 6)                       
Out[2]: 
array([[ 0.,  1.,  2.,  3.,  4.,  5.],
       [ 6.,  7.,  8.,  9., 10., 11.],
       [12., 13., 14., 15., 16., 17.],
       [18., 19., 20., 21., 22., 23.]], dtype=float32)
```

Though I did try running the full test previously with these changes and got errors due to the fact that `cupy.sparse.vstack` and `cupy.sparse.hstack` don't exist, which shows the test is exercising the concatenate code added here. Did not realize at the time you had PR ( https://github.com/cupy/cupy/pull/2665 ) to fix this though.

Any thoughts on what we do with older CuPy versions? Presumably that change will only be admitted to CuPy 8.0.0.

Reply:
That's a great point, and yes the change will be released with CuPy 8.0.0. 

We could import the functions explicitly in `backends.py` and maybe raise a warning / exception if they don't exist. That way the user isn't left wondering why the function being called doesn't exist. I'm interested in hearing your thoughts here as well.

Reply:
Yeah that seems sensible. I guess it needs to be an exception if we have no fallback behavior.